### PR TITLE
testing: Add framework and title test

### PR DESCRIPTION
**Pull request by ZJvandeWeg**
## Description

There's a few bugs with titles, this change adds some plumbing to test the output instead of manually verifying.

### CODE REVIEW

**Comment by joepavitt** [comment]
Tests currently failing

**Comment by ZJvandeWeg** [comment]
@joepavitt Fixed!

File: test/title.spec.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+const assert = require('assert');
+const fs = require('fs');
+const path = require('path');
+
+
```
Comment (ZJvandeWeg) [inline test/title.spec.js:5]:
```suggestion

```

File: test/title.spec.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+describe("Page titles", async function() {
+	it("displays the homepage title", async function() {
+		var data;
+		data = fs.readFileSync(path.resolve(__dirname, "../_site/index.html"), 'utf8');
```
Comment (ZJvandeWeg) [inline test/title.spec.js:9]:
```suggestion
		const data = fs.readFileSync(path.resolve(__dirname, "../_site/index.html"), 'utf8');
```

File: test/title.spec.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+		var xpath = require('xpath');
+		var dom = require('xmldom').DOMParser;
+
+		var doc = new dom().parseFromString(data)
+		var nodes = xpath.select("//title", doc)
```
Comment (ZJvandeWeg) [inline test/title.spec.js:15]:
```suggestion
		var nodes = xpath.select("//title", new dom().parseFromString(data))
```

File: test/title.spec.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+const path = require('path');
+
+
+describe("Page titles", async function() {
+	it("displays the homepage title", async function() {
+		var data;
+		data = fs.readFileSync(path.resolve(__dirname, "../_site/index.html"), 'utf8');
+
+		var xpath = require('xpath');
+		var dom = require('xmldom').DOMParser;
```
Comment (ZJvandeWeg) [inline test/title.spec.js:12]:
```suggestion
		const xpath = require('xpath');
		const dom = require('xmldom').DOMParser;
```

File: scripts/copy_docs.js [hunk 5]
```diff
@@ -41,6 +41,10 @@ async function copyFiles (src, dest, version) {
 }
 
 (async () => {
+	  if(process.env.MANUAL_DOCS_COPY) {
+			return
+		}
```
Comment (ZJvandeWeg) [inline scripts/copy_docs.js:46]:
```suggestion
    if(process.env.MANUAL_DOCS_COPY) {
        return
    }
```

File: scripts/copy_handbook.js [hunk 6]
```diff
@@ -40,6 +40,10 @@ async function copyFiles (src, dest) {
 }
 
 (async () => {
+	  if(process.env.MANUAL_HANDBOOK_COPY) {
+			return
+		}
```
Comment (ZJvandeWeg) [inline scripts/copy_handbook.js:45]:
```suggestion
    if(process.env.MANUAL_HANDBOOK_COPY) {
        return
    }
```

**Comment by joepavitt** [comment]
anything we can do to supress the warning?

Screenshot 2023-02-21 at 09 41 47

**Comment by ZJvandeWeg** [comment]
@joepavitt Done in b9804f2b14f44a5a6a7eaa566a93e830a6848413

**Comment by joepavitt** [comment]
resolved conflict, checked and test locally, all working fine - just waiting for remote tests to run and validate before merging

**Review by joepavitt — APPROVED** [review]

**Pull request merged**